qiskit/providers/ibmq/ibmqbackendservice.py

```diff
@@ -198,14 +204,27 @@ def jobs(
         if job_name:
             api_filter['name'] = {"regexp": job_name}
 
-        if start_datetime and end_datetime:
-            api_filter['creationDate'] = {
-                'between': [start_datetime.isoformat(), end_datetime.isoformat()]
-            }
-        elif start_datetime:
-            api_filter['creationDate'] = {'gte': start_datetime.isoformat()}
-        elif end_datetime:
-            api_filter['creationDate'] = {'lte': end_datetime.isoformat()}
+        # TODO: Remove when decided the warning is no longer needed.
+        if start_datetime or end_datetime:
+            warnings.warn('The parameters `start_datetime` and `end_datetime` are expected '
+                          'to be in local time now, rather than UTC. If the datetime is in '
+                          'UTC, it will be converted to local time.', stacklevel=2)
+
+            # If the datetime timezone info is not UTC, then convert the datetime into UTC.
+            # Note: datetime objects without a timezone are considered to be in local time.
+            if start_datetime and (start_datetime.tzinfo not in (tz.UTC, timezone.utc)):
```

Comment:
There are other implementations of `tzinfo` that people might use (e.g. `pytz`). Rather than checking if it's one in the set, it's probably more accurate to check for `utcoffset()`.